src/server/job_func.c

```diff
@@ -566,6 +566,18 @@ job_free(job *pj)
 	reliable_job_node_free(&pj->ji_failed_node_list);
 	reliable_job_node_free(&pj->ji_node_list);
 
+	if (pj->ji_bg_hook_task) {
+		mom_process_hooks_params_t *php;
+		php = pj->ji_bg_hook_task->wt_parm2;
+		if (php->hook_output) {
+			free(php->hook_output->reject_errcode);
+			free(php->hook_output);
+		}
+		free(php->hook_input);
+		free(php);
+		pj->ji_bg_hook_task->wt_parm2 = NULL;
```

Comment:
I think you should just delete the task altogether here, like adding delete_task(pj-JI_bg_hook_task). Otherwise, main mom would continue to see the task, process it, and although upon seeing pj->ji_bg_hook_task->wt_parm2 (under post_run_hook()) being NULL, it would return immediately. But this would result in a mom_logs message "missing hook params argument to event". So I think it's better to delete the task now.